usnistgov/oar-pdr
# Get Angular unit tests working again

Angular unit (and e2e) tests have not been working for sometime for the landing page service, and running them has been disabled in the TravisCI configuration.  The primary goal of this PR is to fix the problems preventing their execution.  A secondary goal is to update and stream-line the angular-cli/universal environment files to improve maintainability of our angular codebase.  This PR does not address the e2e tests (these have not even been tested).  

The changes introduced in this PR include:
   * changes to files that are part of the angular and karma frameworks
   * updates to the `package.json` file, adding dependencies or updating versions; modules apparently no longer needed have been removed.  
   * changes to `*.spec.ts` files to get the units tests run and pass
   * changes to angular code necessary to get it run under operational conditions (namely, under docker).  Inparticular, these enable datacart-related code to run without error during server-side rendering.  

Currently, there are seven unit tests running, and these have been "turned on" in the TravisCI configuration.  There are others that have been commented out.  I attempted to uncomment a few of these, but I found that they need to be updated in order to pass.

To test:
   * build via `npm run build:ssr`; ensure it completes successfully,
   * run `npm test` and ensure all tests pass,
   * run in standalone server mode (`npm run serve` or `npm run serve:ssr`); ensure pages look okay,
   * run under oar-docker,
   * inspect landing and about pages, look for any rendering errors or oddities.
